### PR TITLE
Add ProjectionService support to all plugins/visualizers

### PR DESCRIPTION
**Pull request by liorze**
Ready for review.

### CODE REVIEW

File: src/app/@ansyn/plugins/openlayers/open-layers-map/openlayers-map/openlayers-map.ts [hunk 1]
```diff
@@ -273,64 +295,66 @@ export class OpenLayersMap extends IMap<OLMap> {
...
 	public setCenter(center: GeoJSON.Point, animation: boolean) {
-		const view = this._mapObject.getView();
-		const projection = view.getProjection();
-		const olCenter = proj.transform([center.coordinates[0], center.coordinates[1]], 'EPSG:4326', projection);
-		if (animation) {
-			this.flyTo(olCenter);
-		} else {
-			view.setCenter(olCenter);
-		}
+		this.projectionService.projectAccuratelyToImage(center, this).subscribe(projectedCenter => {
```
[the remark this answers is not in the document]
Comment (yairtawil) [reply src/app/@ansyn/plugins/openlayers/open-layers-map/openlayers-map/openlayers-map.ts]: why setCenter get unProjected center, and getCenter return projected center?

File: src/app/@ansyn/plugins/openlayers/open-layers-map/openlayers-map/openlayers-map.ts [hunk 2]
```diff
@@ -273,64 +295,66 @@ export class OpenLayersMap extends IMap<OLMap> {
...
+			.map((collection: FeatureCollection<GeometryObject>) => collection.features[0].geometry as Polygon);
 	}
 
 
 	fitRotateExtent(map: OLMap, extentFeature: CaseMapExtentPolygon) {
-		const view: ol.View = map.getView();
-		const size = map.getSize();
-		const projection = view.getProjection();
-		extentFeature = ExtentCalculator.transform(proj.transform.bind(proj), extentFeature, 'EPSG:4326', projection);
+		const collection: GeoJSON.FeatureCollection<Polygon> = turf.featureCollection([turf.feature(extentFeature)]);
```
Comment (yairtawil) [inline src/app/@ansyn/plugins/openlayers/open-layers-map/openlayers-map/openlayers-map.ts]: return `Observable<boolean>` ?

File: src/app/@ansyn/plugins/openlayers/open-layers-map/openlayers-map/openlayers-map.ts [hunk 3]
```diff
@@ -273,64 +295,66 @@ export class OpenLayersMap extends IMap<OLMap> {
...
+				const resolution = ExtentCalculator.calcResolution(geoJsonExtent, map.getSize(), rotation);
+
+				view.setCenter(center);
+				view.setRotation(rotation);
+				view.setResolution(resolution);
+
+			});
 	}
 
 	public setPosition(position: CaseMapPosition, view: View = this.mapObject.getView()): void {
```
Comment (yairtawil) [inline src/app/@ansyn/plugins/openlayers/open-layers-map/openlayers-map/openlayers-map.ts]: return Observable.. who called setPosition should subscribe it

File: src/app/@ansyn/plugins/openlayers/open-layers-north-calculations/plugin/north-calculations-plugin.ts [hunk 4]
```diff
@@ -35,21 +40,20 @@ export class NorthCalculationsPlugin implements IMapPlugin {
 		this.onDisposedEvent.emit();
 	}
 
-	getCorrectedNorthOnce(mapObject): Promise<INorthData> {
+	getCorrectedNorthOnce(iMap): Observable<INorthData> {
```
Comment (yairtawil) [inline src/app/@ansyn/plugins/openlayers/open-layers-north-calculations/plugin/north-calculations-plugin.ts]: type iMap.... IMap "<"olMap">"

File: src/app/@ansyn/plugins/openlayers/open-layer-visualizers/tools/measure-distance.visualizer.ts [hunk 5]
```diff
@@ -142,7 +145,7 @@ export class MeasureDistanceVisualizer extends EntitiesVisualizer {
 					id: UUID.UUID(),
 					featureJson
 				};
-				this.addOrUpdateEntities([newEntity]);
+				this.addOrUpdateEntities([newEntity]).subscribe();
```
[the remark this answers is not in the document]
Comment (liorze) [reply src/app/@ansyn/plugins/openlayers/open-layer-visualizers/tools/measure-distance.visualizer.ts]: Yap, I really think we should change the `ProjectionService` API.

File: src/app/@ansyn/plugins/openlayers/open-layers-north-calculations/plugin/north-calculations-plugin.ts (hunk 4 above)
Comment (liorze) [reply src/app/@ansyn/plugins/openlayers/open-layers-north-calculations/plugin/north-calculations-plugin.ts]: Done

File: src/app/@ansyn/plugins/openlayers/open-layer-visualizers/tools/measure-distance.visualizer.ts [hunk 6]
```diff
@@ -102,11 +103,13 @@ export class MeasureDistanceVisualizer extends EntitiesVisualizer {
 		this.geoJsonFormat = new GeoJSON();
 	}
 
-	onResetView() {
-		super.onResetView();
+	onResetView(): Observable<boolean> {
+		let obs = super.onResetView();
```
[the remark this answers is not in the document]
Comment (liorze) [reply src/app/@ansyn/plugins/openlayers/open-layer-visualizers/tools/measure-distance.visualizer.ts]: Done

File: src/app/@ansyn/core/utils/extent-calculator.ts [hunk 7]
```diff
@@ -1,24 +1,24 @@
...
+	static extentToPolygon(extent: CaseMapExtent): GeoJSON.Feature<GeoJSON.Polygon> {
 		const minX = extent[0];
 		const minY = extent[1];
 		const maxX = extent[2];
 		const maxY = extent[3];
 
 		const coordinates = [[minX, minY], [minX, maxY], [maxX, maxY], [maxX, minY], [minX, minY]];
 
-		return { type: 'Polygon', coordinates: [coordinates] };
+		return turf.polygon([coordinates]);
```
Comment (yairtawil) [inline src/app/@ansyn/core/utils/extent-calculator.ts]: it's a feature! should be geometry

File: src/app/@ansyn/map-facade/effects/map.effects.ts [hunk 8]
```diff
@@ -313,11 +328,14 @@ export class MapEffects {
...
 		.withLatestFrom(this.store$.select(mapStateSelector))
 		.filter(([{ payload }, { mapsList }]: [ImageryCreatedAction, IMapState]) => !MapFacadeService.mapById(mapsList, payload.id).data.position)
-		.mergeMap(([{ payload }, mapState]: [ImageryCreatedAction, IMapState]) => {
-			const actions = [];
+		.switchMap(([{ payload }, mapState]: [ImageryCreatedAction, IMapState]) => {
+				const activeMap = MapFacadeService.activeMap(mapState);
+				const communicator = this.communicatorsService.provide(payload.id);
+				return communicator.setPosition(activeMap.data.position).map(() => [{ payload }, mapState]);
+			})
+		.switchMap(([{ payload }, mapState]: [ImageryCreatedAction, IMapState]) => {
```
Comment (yairtawil) [inline src/app/@ansyn/map-facade/effects/map.effects.ts]: switchMap ? are you sure?
```typescript
return actions
```

File: src/app/@ansyn/plugins/openlayers/open-layers-map/openlayers-map/openlayers-map.ts [hunk 9]
```diff
@@ -109,33 +111,50 @@ export class OpenLayersMap extends IMap<OLMap> {
...
+	initMap(element: HTMLElement, layers: any, position?: CaseMapPosition): Observable<boolean> {
 		const [mainLayer] = layers;
 		const view = this.createView(mainLayer);
 		const coordinateFormat: ol.CoordinateFormatType = (coords: ol.Coordinate): string => coords.map((num) => +num.toFixed(4)).toString();
 		this._mapObject = new OLMap({
 			target: element,
 			layers,
 			renderer: 'canvas',
-			controls: [new ScaleLine(), new MousePosition({ projection: 'EPSG:4326', coordinateFormat  })],
+			controls: [new ScaleLine(), new OpenLayersMousePositionControl({ projection: 'EPSG:4326', coordinateFormat  },
```
Comment (yairtawil) [inline src/app/@ansyn/plugins/openlayers/open-layers-map/openlayers-map/openlayers-map.ts]: 👏 👏 👏

File: src/app/@ansyn/plugins/openlayers/open-layers-map/openlayers-map/openlayers-mouseposition-control.ts [hunk 10]
```diff
@@ -0,0 +1,56 @@
+import MousePosition from 'ol/control/mouseposition';
+import { Point } from 'geojson';
+import { Observable } from 'rxjs/Observable';
+import * as turf from '@turf/turf';
+import { Subscription } from 'rxjs/Subscription';
+import Projection from 'ol/proj/projection';
+
+export class OpenLayersMousePositionControl extends MousePosition {
```
Comment (yairtawil) [inline src/app/@ansyn/plugins/openlayers/open-layers-map/openlayers-map/openlayers-mouseposition-control.ts]: wow

File: src/app/@ansyn/core/utils/extent-calculator.ts (hunk 7 above)
Comment (liorze) [reply src/app/@ansyn/core/utils/extent-calculator.ts]: I'm expecting a feature, look at the usages of this function.

File: src/app/@ansyn/map-facade/effects/map.effects.ts (hunk 8 above)
Comment (liorze) [reply src/app/@ansyn/map-facade/effects/map.effects.ts]: I don't understand this comment.

**Pull request merged**